File: ts-bun/file-tasks.ts
// file-tasks.ts - File I/O operations
import { readFileSync, writeFileSync, existsSync } from 'fs';

// Search for a word in CSV file and write results
export async function searchCsvFile(filePath: string, searchWord: string): Promise<void> {
  // Check if file exists
  if (!existsSync(filePath)) {
    const message = `CSV file not found: ${filePath}\nSkipping CSV search.`;
    console.log(`  ⚠️  ${message}`);
    writeFileSync('search_results.txt', message);
    return;
  }
  
  // Read the entire CSV file
  const content = readFileSync(filePath, 'utf-8');
  
  // Count occurrences of the search word (case-insensitive)
  const regex = new RegExp(searchWord, 'gi');
  const matches = content.match(regex);
  const count = matches ? matches.length : 0;
  
  // Prepare result message
  const resultMessage = count > 0 
    ? `Found ${count} matches for "${searchWord}"`
    : `No matches found for "${searchWord}"`;
  
  // Write results to file
  writeFileSync('search_results.txt', resultMessage);
  
  console.log(`  ✓ ${resultMessage}`);
}

// Additional file utility: Read lines from CSV
export function readCsvLines(filePath: string): string[] {
  if (!existsSync(filePath)) {
    return [];
  }
  
  const content = readFileSync(filePath, 'utf-8');
  return content.split('\n').filter(line => line.trim().length > 0);
}

File: python/file_tasks.py
"""
file_tasks.py - File I/O operations
"""

import os
import re
from typing import List


def search_csv_file(file_path: str, search_word: str) -> None:
    """Search for a word in CSV file and write results"""

    # Check if file exists
    if not os.path.exists(file_path):
        message = f"CSV file not found: {file_path}\nSkipping CSV search."
        print(f"  ⚠️  {message}")
        with open("search_results.txt", "w") as f:
            f.write(message)
        return

    # Read the entire CSV file
    with open(file_path, "r", encoding="utf-8") as f:
        content = f.read()

    # Count occurrences of the search word (case-insensitive)
    pattern = re.compile(search_word, re.IGNORECASE)
    matches = pattern.findall(content)
    count = len(matches)

    # Prepare result message
    if count > 0:
        result_message = f'Found {count} matches for "{search_word}"'
    else:
        result_message = f'No matches found for "{search_word}"'

    # Write results to file
    with open("search_results.txt", "w") as f:
        f.write(result_message)

    print(f"  ✓ {result_message}")


def read_csv_lines(file_path: str) -> List[str]:
    """Read lines from CSV file"""
    if not os.path.exists(file_path):
        return []

    with open(file_path, "r", encoding="utf-8") as f:
        lines = [line.strip() for line in f if line.strip()]

    return lines
